tools/config_utils.py

```diff
@@ -0,0 +1,5 @@
+import sys
+from pathlib import Path
+import numpy as np
+outfile = Path(sys.argv[1])
+outfile.write_text(Path(str(np.__config__.__file__)).read_text())
```

Comment:
This is not correct, it will just copy the numpy `__config__.py` file, rather than write out SciPy's build-time config settings. It's actually hard to do this correctly, because the file is generated by `config.make_config_py()` in `scipy/setup.py` - which relies on an instance of `numpy.distutils.Configuration` which we cannot access with a Meson build.

So, let's leave it like this for now (it's not blocking for this PR), but add a descriptive comment about what is going on here.

Reply:
The implementation of this method is at https://github.com/numpy/numpy/blob/main/numpy/distutils/misc_util.py#L2113, let's link that so it's easy to find back later.

Reply:
Added a detailed comment.